src/test/java/teammates/test/cases/logic/CoursesLogicTest.java

```diff
@@ -775,26 +778,34 @@ private void testHasIndicatedSections() throws Exception {
 
     private void testCreateCourse() throws Exception {
 
-        /*Explanation:
-         * The SUT (i.e. CoursesLogic::createCourse) has only 1 path. Therefore, we
-         * should typically have 1 test cases here.
-         */
         ______TS("typical case");
 
         CourseAttributes c = CourseAttributes
-                .builder("Computing101-fresh", "Basic Computing", "Asia/Singapore")
+                .builder("Computing101-fresh", "Basic Computing", ZoneId.of("Asia/Singapore"))
                 .build();
-        coursesLogic.createCourse(c.getId(), c.getName(), c.getTimeZone());
+        coursesLogic.createCourse(c.getId(), c.getName(), c.getTimeZone().getId());
         verifyPresentInDatastore(c);
         coursesLogic.deleteCourseCascade(c.getId());
         ______TS("Null parameter");
 
         try {
-            coursesLogic.createCourse(null, c.getName(), c.getTimeZone());
+            coursesLogic.createCourse(null, c.getName(), c.getTimeZone().getId());
             signalFailureToDetectException();
         } catch (AssertionError e) {
             assertEquals("Non-null value expected", e.getMessage());
         }
+        ______TS("Invalid time zone");
+
+        String invalidTimeZone = "Invalid Timezone";
```

Comment:
Test invalid ID too together with invalid time zone to check that both error messages are included.